client/landing/jetpack-cloud/components/sidebar/index.jsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import ExpandableSidebarMenu from 'layout/sidebar/expandable';
+import Sidebar from 'layout/sidebar';
+import SidebarFooter from 'layout/sidebar/footer';
+import SidebarItem from 'layout/sidebar/item';
+import SidebarMenu from 'layout/sidebar/menu';
+import SidebarRegion from 'layout/sidebar/region';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class JetpackCloudSidebar extends Component {
+	static propTypes = {
+		path: PropTypes.string.isRequired,
+	};
+
+	state = {
+		expandedSections: this.props.path.match( /^(\/?[^/]*)/gi ), // @todo: Restore the sections state after page refresh
+	};
+
+	/**
+	 * Toggle an expandable section.
+	 *
+	 * @param {string} path Section path
+	 */
+	toggleSection( path ) {
+		const expandedSections = this.isExpanded( path )
+			? this.state.expandedSections.filter( item => item !== path )
+			: [ ...this.state.expandedSections, path ];
+		this.setState( { expandedSections } );
+	}
+
+	/**
+	 * Check if a menu is expanded.
+	 *
+	 * @param   {string}  path Section path
+	 * @returns {boolean}      True if section is expanded
+	 */
+	isExpanded( path ) {
+		return this.state.expandedSections.includes( path );
+	}
+
+	/**
+	 * Check if a menu item is selected.
+	 *
+	 * @param   {string}  path Menu item path
+	 * @returns {boolean}      True if menu item is selected
+	 */
+	isSelected( path ) {
+		return this.props.path === path;
+	}
+
+	handleExpandableMenuClick( path ) {
+		return () => this.toggleSection( path );
+	}
+
+	onNavigate = () => {
+		window.scrollTo( 0, 0 );
+	};
```

Comment:
Just curious on this: why does only this function end with a `;` semicolon?

Reply:
The `onNavigate` is a class *property* declaration and as such should be terminated with a semicolon. (To be more clear: it's a class property which gets assigned an arrow function).

The other functions are class *method* declarations and they don't need a semicolon at the end.

I may be wrong about that, though 🙂

Reply:
Ahhh, that makes more sense now. Sorry for hijacking your PR to learn about JS 🤣 Thank you.